商品詳細表示機能

nanaiwamoto opened the pull request:
#what
商品詳細表示機能の実装をしました。
確認をお願いします。

また、
実施要項の注意事項に
「機能毎に、commitとpushを行うこと。」とありますが、これば「商品詳細表示機能」では一回のcommitとpushを行うという意味でしょうか。
よろしくお願いいたします。

#why
表示機能がLGTMでしたら次の実装を行います。

▢以下に必要事項の動画のリンクを記載します。

[ログイン状態且つ、自身が出品した販売中商品の商品詳細ページへ遷移した動画](https://gyazo.com/983d5cf7812f08ebbe7c6332fd98b37e)

[ログイン状態且つ、自身が出品していない販売中商品の商品詳細ページへ遷移した動画](https://gyazo.com/7b794cba5721a1e9a98e93812af5ee1a)

ログイン状態で、売却済み商品の商品詳細ページへ遷移した動画（現段階で商品購入機能の実装が済んでいる場合）
→現段階で実装が済んでいないため添付していません。

[ログアウト状態で、商品詳細ページへ遷移した動画](https://gyazo.com/7a519d02e501b4e7a9b9f3fcd2080fd9)

nanaiwamoto committed:
商品詳細表示機能

OhyaYouki reviewed (commented):
レビュー依頼ありがとうございます！

こちらのプルリクエストにてコンフリクトが起きていますので、まずはこちらを解消してください。
githubのブラウザエディタで「Resolve conflicts」をクリックして解消しましょう。
ご不明点などございましたら、コネクトでご質問ください。

その後、再度レビュー依頼フォームにてご提出ください。
お手数をお掛けいたしますが、よろしくお願いいたします。

nanaiwamoto committed:
Merge branch 'main' into 商品詳細表示機能

nanaiwamoto commented:
コンフリクトの解消をおこないました！
確認をお願いします。

#what
商品詳細表示機能の実装をしました。
確認をお願いします。

また、
実施要項の注意事項に
「機能毎に、commitとpushを行うこと。」とありますが、これば「商品詳細表示機能」では一回のcommitとpushを行うという意味でしょうか。
よろしくお願いいたします。

#why
表示機能がLGTMでしたら次の実装を行います。

▢以下に必要事項の動画のリンクを記載します。

[ログイン状態且つ、自身が出品した販売中商品の商品詳細ページへ遷移した動画](https://gyazo.com/983d5cf7812f08ebbe7c6332fd98b37e)

[ログイン状態且つ、自身が出品していない販売中商品の商品詳細ページへ遷移した動画](https://gyazo.com/7b794cba5721a1e9a98e93812af5ee1a)

ログイン状態で、売却済み商品の商品詳細ページへ遷移した動画（現段階で商品購入機能の実装が済んでいる場合）
→現段階で実装が済んでいないため添付していません。

[ログアウト状態で、商品詳細ページへ遷移した動画](https://gyazo.com/7a519d02e501b4e7a9b9f3fcd2080fd9)

wad-sho commented on app/views/items/show.html.erb:
```diff
@@ -4,67 +4,67 @@
+          <td class="detail-value"><%= @item.shopping_day.name %></td>
         </tr>
       </tbody>
     </table>
```
※該当箇所にコメントができないためこちらに失礼いたします。

詳細ページ最下部の以下記述においても保存した商品のカテゴリー名が表示されるようにしましょう。
また、実装後は上下のコメントアウトの削除もお忘れなくお願いします。

`<%='商品のカテゴリー名'%>をもっと見る`

wad-sho commented on app/views/items/show.html.erb:
```diff
@@ -4,67 +4,67 @@
-    <%# //ログインしているユーザーと出品しているユーザーが、同一人物の場合と同一人物ではない場合で、処理を分けましょう %>
+  <% end %>
+<% end %>
+<%# //ログインしているユーザーと出品しているユーザーが、同一人物の場合と同一人物ではない場合で、処理を分けましょう %>
```
```
<%# ログインしているユーザーと出品しているユーザーが、同一人物の場合と同一人物ではない場合で、処理を分けましょう %>
<%# //ログインしているユーザーと出品しているユーザーが、同一人物の場合と同一人物ではない場合で、処理を分けましょう %>
```
上記コメントアウトは商品詳細表示機能の実装案内ですので削除しましょう。

wad-sho commented on app/models/item.rb:
```diff
@@ -2,6 +2,7 @@ class Item < ApplicationRecord
   belongs_to :user
   # 商品購入機能実装までOUT has_one :order
   has_one_attached :image
+  has_one :order
```
orderとのアソシエーションは商品購入機能にて実装します。
現時点は削除かコメントアウトをお願いします。

wad-sho commented on app/models/order.rb:
```diff
@@ -1,5 +1,6 @@
-  # belongs_to :item
+
+  belongs_to :item
+  belongs_to :user
```
orderとのアソシエーションは商品購入機能にて実装します。
現時点は削除かコメントアウトをお願いします。

wad-sho commented on app/models/user.rb:
```diff
@@ -1,7 +1,8 @@
-  # has_many :orders
+
   has_many :items
+  has_many :orders
```
orderとのアソシエーションは商品購入機能にて実装します。
現時点は削除かコメントアウトをお願いします。

wad-sho reviewed (commented):
実装お疲れ様です！レビューのご依頼ありがとうございます。

コメントいたしましたので、ご確認・修正の後、再度レビューのご依頼お願いいたします！

> 「機能毎に、commitとpushを行うこと。」とありますが、これば「商品詳細表示機能」では一回のcommitとpushを行うという意味でしょうか。
いえ、ブランチの機能ごとに一度のコミットプッシュを行う、という意味ではございません。コミットの粒度はこまめに行いましょう。

※レビューではご質問対応は不可となります。今後質問対応はできませんのでご認識お願いいたします。

nanaiwamoto committed:
商品詳細表示機能

（プルリクエスト後の修正点）
詳細ページ最下部の以下記述においても保存した商品のカテゴリー名が表示されるようにしましょう。
また、実装後は上下のコメントアウトの削除もお忘れなくお願いします。

<%='商品のカテゴリー名'%>をもっと見る

nanaiwamoto committed:
Merge branch '商品詳細表示機能' of https://github.com/nanaiwamoto/furima-41245 into 商品詳細表示機能

nanaiwamoto committed:
商品詳細表示機能

nanaiwamoto commented:
確認ありがとうございます。
いただいた箇所訂正しました。
確認お願いします。　

動画も以下に再掲載します。
▢以下に必要事項の動画のリンクを記載します。

[ログイン状態且つ、自身が出品した販売中商品の商品詳細ページへ遷移した動画](https://gyazo.com/983d5cf7812f08ebbe7c6332fd98b37e)

[ログイン状態且つ、自身が出品していない販売中商品の商品詳細ページへ遷移した動画](https://gyazo.com/7b794cba5721a1e9a98e93812af5ee1a)

ログイン状態で、売却済み商品の商品詳細ページへ遷移した動画（現段階で商品購入機能の実装が済んでいる場合）
→現段階で実装が済んでいないため添付していません。

[ログアウト状態で、商品詳細ページへ遷移した動画](https://gyazo.com/7a519d02e501b4e7a9b9f3fcd2080fd9)

wad-sho commented on app/views/items/show.html.erb:
```diff
@@ -4,67 +4,66 @@
-
-    <%= link_to "商品の編集", "#", method: :get, class: "item-red-btn" %>
+   <%# ログインしているユーザーと出品しているユーザーが、同一人物の場合と同一人物ではない場合で、処理を分けましょう %>
+<% if user_signed_in? %>
```
`<%# ログインしているユーザーと出品しているユーザーが、同一人物の場合と同一人物ではない場合で、処理を分けましょう %>`

こちらは実装案内です。
残っていることは好ましくありませんので削除をお願いします。

wad-sho commented on app/views/items/show.html.erb:
```diff
@@ -4,67 +4,66 @@
-
-
-    <%# 商品が売れていない場合はこちらを表示しましょう %>
+    <%= link_to "削除", item_path(@item), data: {turbo_method: :delete}, class:"item-destroy" %>
```
編集、および削除ボタンのパスを"#"に戻してください。

それぞれの機能に入った際に実装しましょう。

wad-sho reviewed (commented):
実装お疲れ様です！レビューのご依頼ありがとうございます。
後少しです！

コメントいたしましたので、ご確認・修正の後、再度レビューのご依頼お願いいたします！

nanaiwamoto commented:
確認ありがとうございます。
確認をお願いいたします。

wad-sho reviewed (commented):
レビュー依頼ありがとうございます！

こちらのプルリクエストですが、現在修正点の差分があがっていないようです。
コミットとプッシュのし忘れがないか、ご確認お願いします！

nanaiwamoto committed:
商品詳細表示機能

nanaiwamoto commented:
コミットとプッシュのし忘れでした！大変申し訳ございません・・・

確認をお願いいたします。

wad-sho reviewed (commented):
レビューのご依頼ありがとうございます。

実装内容を確認させていただきました。
**商品詳細表示機能、LGTMとさせていただきます。実装お疲れ様でした！**

このブランチをマージし、mainブランチでプルしましょう。
内容が反映されていることを確認し、次の実装・学習に入りましょう！
引き続き応援しております！

nanaiwamoto commented:
ありがとうございました😭！
引き続きよろしくお願いいたします。
